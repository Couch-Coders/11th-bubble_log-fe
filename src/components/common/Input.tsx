import React from 'react';
import styled from 'styled-components';

const Container = styled.input`
  border-radius: 0.5rem;
  padding: 0.5rem;
`;

interface Props extends React.InputHTMLAttributes<HTMLInputElement> {}

const Input: React.FC<Props> = () => {
  return <Container />;
};

export default React.memo(Input);
