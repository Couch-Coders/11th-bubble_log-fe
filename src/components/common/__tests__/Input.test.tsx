import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

import Input from '@components/common/Input';

describe('Input', () => {
  it('컴포넌트가 올바르게 렌더링되어야 합니다.', () => {
    const { container } = render(<Input value="default value" />);
    const input = screen.getByDisplayValue('default value');

    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('placeholder가 올바르게 렌더링되어야 합니다.', () => {
    render(<Input placeholder="default placeholder" />);

    const input = screen.getByPlaceholderText('default placeholder');
    expect(input).toBeInTheDocument();
  });

  it('컴포넌트의 값이 올바르게 변경되어야 합니다.', () => {
    render(<Input placeholder="default placeholder" />);

    const input: HTMLInputElement = screen.getByPlaceholderText(
      'default placeholder',
    );
    fireEvent.change(input, { target: { value: 'test value' } });
    expect(input.value).toBe('test value');
  });
});
