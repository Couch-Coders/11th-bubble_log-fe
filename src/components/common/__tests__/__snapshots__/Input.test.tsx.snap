// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 컴포넌트가 올바르게 렌더링되어야 합니다. 1`] = `
<div>
  <input
    class="sc-bcXHqe dsxYIO"
    value="default value"
  />
</div>
`;
